arangodb-net-standard/Serialization/JsonNetApiClientSerialization.cs

```diff
@@ -32,7 +32,18 @@ public virtual T DeserializeFromStream<T>(Stream stream)
             {
                 var js = new JsonSerializer();
 
-                T result = js.Deserialize<T>(jtr);
+                T result;
+                try
+                {
+                    result = js.Deserialize<T>(jtr);
+                }
+                catch (JsonSerializationException e)
+                {
+                    if (e.InnerException?.Message == "Null object cannot be converted to a value type.")
+                        throw new Exception($"Failed to Deserialize the Arango result to type {typeof(T).Name}. Arango returned [null] as value, which cannot be converted to your valuetype. Consider using a Nullable valuetype.", e);
+                    else
+                        throw new Exception($"Failed to Deserialize the Arango result to type {typeof(T).Name}. See InnerException for more details.", e);
```

Comment:
> I don't understand how it is supposed to improve interpretation/catch for the consumer.

Suddenly I was confronted with a "Error converting value {null} to type 'System.Int32'". 
I was not expecting NULL from Arango. So I was unsure about what triggered this error. As a developer, you do not know the ins and outs of the client. For all I know it could have been a bug in the client. 
So this is what I need as debugger: undoubtable information about the context of the error.